client/service/src/genesis.rs

```diff
@@ -21,21 +21,26 @@ use sp_runtime::traits::{Block as BlockT, Hash as HashT, Header as HeaderT, Zero
 /// Generate the genesis block from a given ChainSpec.
 pub fn generate_genesis_block<Block: BlockT>(
 	chain_spec: &Box<dyn ChainSpec>,
+	genesis_state_version: sp_runtime::StateVersion,
```

Comment:
Can we not fetch the runtime version from the code in the genesis state? As we do this in Substrate?

Reply:
That was my first idea, but current code need an executor (for compatibility I think), so ends up being really easier to use this way.
You think it is a must have, or can we keep it this way (seems only use from commands at this point)?

Reply:
Hmm, this is not really future proof.